dev/ipynb_to_gallery.py

```diff
@@ -0,0 +1,65 @@
+"""
+This python script is inspired from https://gist.github.com/chsasank/7218ca16f8d022e02a9c0deb94a310fe .
+
+Convert jupyter notebook to sphinx gallery python script.
+
+Usage: python ipynb_to_gallery.py <notebook.ipynb> (Optional)<output.py>
+"""
+import json
+import pypandoc as pdoc
+
+
+def convert_ipynb_to_gallery(input_file_name, output_file_name=None):
+    """
+    Convert jupyter notebook to sphinx gallery python script.
+
+    parameters
+    ----------
+    input_file_name: str
+        Path to the jupyter notebook file.
+    output_file_name: str
+        Path to the output python file. If not provided, the output file name is the same as the input file name.
+    """
+    python_file = ""
+
+    nb_dict = json.load(open(input_file_name))
+    cells = nb_dict["cells"]
+
+    for i, cell in enumerate(cells):
+        if i == 0:
+            assert cell["cell_type"] == "markdown", "First cell has to be markdown"
+
+            md_source = "".join(cell["source"])
+            rst_source = pdoc.convert_text(md_source, "rst", "md").replace("``", "`")
+            python_file = '"""\n' + rst_source + '\n"""'
+        else:
+            if cell["cell_type"] == "markdown":
+                md_source = "".join(cell["source"])
+                rst_source = pdoc.convert_text(md_source, "rst", "md").replace(
+                    "``", "`"
+                )
+                commented_source = "\n".join(["# " + x for x in rst_source.split("\n")])
+                python_file = (
+                    python_file + "\n\n\n" + "#" * 70 + "\n" + commented_source
+                )
+            elif cell["cell_type"] == "code":
+                source = "".join(cell["source"])
+                python_file = python_file + "\n" * 2 + source
+
+    python_file = python_file.replace("\n%", "\n# %")
+    if output_file_name is None:
+        output_file_name = input_file_name
+        open(output_file_name.replace(".ipynb", ".py"), "w").write(python_file)
+    else:
+        if not output_file_name.endswith(".py"):
+            output_file_name = output_file_name + ".py"
+        open(output_file_name, "w").write(python_file)
```

Comment:
```suggestion
    if output_file_name is None:
        output_file_name = input_file_name.replace(".ipynb", ".py")
    if not output_file_name.endswith(".py"):
        output_file_name = output_file_name + ".py"
    open(output_file_name, "w").write(python_file)
```
less indentation is usually better

Reply:
Note that you could also use the `gammapy.utils.script.make_path` function which allows you to use `pathlib.Path` objects with e.g. `Path.write_text(python_file)` or `Path.with_suffix(".py")`

Reply:
Yes it's true that I could have used that. 
Now that it is written like this, do you see any reason why we should switch to `Path`?